### PR TITLE
Vendor dependencies + migration of deprecated API groups

### PR DESCRIPTION
**Pull request by glitchcrab**
Towards giantswarm/giantswarm/issues/6013

### CODE REVIEW

**Comment by sslavic** [comment]
I've rebased this branch to include latest master changes and vendored in all of the dependencies. The PR now also includes migration of some of the deprecated API groups. This all slims down k8s+helm update https://github.com/giantswarm/draughtsman/pull/95 PR.

PR is open for review.

**Review by rossf7 — APPROVED** [review]
LGTM

File: helm/draughtsman-chart/templates/rbac.yaml [hunk 1]
```diff
@@ -35,6 +35,7 @@ rules:
       - clusterroles
     verbs:
       - bind
+      - create
```
Comment (kopiczko) [inline helm/draughtsman-chart/templates/rbac.yaml]: Are you sure we need new verb?

**Review by kopiczko — APPROVED** [review]

File: helm/draughtsman-chart/templates/rbac.yaml (hunk 1 above)
Comment (tomahawk28) [reply helm/draughtsman-chart/templates/rbac.yaml]: draughtsman using helmclient, and helmclient have to create `ClusterRole` if cluster does not have a tiller yet. 
https://github.com/giantswarm/helmclient/blob/386916cdcc22ddd332022e045d26d0e345562116/ensure_tiller_installed.go#L126-L158

Comment (sslavic) [reply helm/draughtsman-chart/templates/rbac.yaml]: Yes, Ji's testing on a cluster uncovered that.

**Comment by sslavic** [comment]
Tested on geckon.

**Pull request merged**